### DISCUSSION TITLE
share java samples of mxnet

### DISCUSSION THREAD

**Discussion by tomoncle** [apache/incubator-mxnet#14943] [Q&A]
used springboot to realize a real-time detection function of uploaded pictures based on SSD model. I hope it can be shared with the sample program of this warehouse, If permitted, I will sort out the submission as soon as possible.

Project demo : https://github.com/tomoncle/mxnet-spring-samples

**Comment by vdantu** [comment]
@mxnet-label-bot add [question]
@lanking520 @frankfliu : Would you be able to take a look at this?

**Comment by git-mosis** [comment]
I run the project in Intellij, load the browser, upload the same image provided in the readme and it's not working like I expect it to. Here's what I'm seeing: 

![uploadImageIssue](https://user-images.githubusercontent.com/22773456/58450860-c5fb3c80-80c5-11e9-8ad8-aff9a36c879a.gif)

Perhaps I'm missing something?

**Comment by tomoncle** [comment]
The image address displayed should be returned by the program itself, not entered manually. 
The upload address should be  http://localhost:9080/files/detectImage

**Comment by git-mosis** [comment]
Hmm, don't see it redirecting and when trying to enter in `/files/detectImage` on a different tab, what's shown is the original upload prompt. Anything else I can try? 

![uploadImageIssue2](https://user-images.githubusercontent.com/22773456/58491926-c0810f00-8124-11e9-9331-baf3c537ef5d.gif)

**Comment by lanking520** [comment]
@tomoncle Thanks for working on this. The scope of the example for Java API is limited to the basic functionalities demo. Your use use case could be applied to https://github.com/awslabs/mxnet-model-server.

**Comment by tomoncle** [comment]
@git-ness Maybe I didn't make it clear that "/files" is a common upload URL, just a reminder that the upload was successful. That's all. Another URL "/files/detectImage" is an interface for image upload and detection. You should use the "/files/detectImage" URL to upload and test. (Note that my test environment is Linux, and if you are MAC os, you need to change your maven configuration)

@lanking520 Thanks.

**Comment by git-mosis** [comment]
Circling back... @lanking520 I thought the idea of the mxnet model server is to serve model files such as the `.json` and `.params` files. Fwiw @tomoncle's example is taking the model & params files, then performs inference and classifies the images themselves in a spring-boot server environment, not just serving the `.json` or `.param` files. 

Perhaps maybe https://github.com/apache/incubator-mxnet/blob/master/example/README.md is a better candidate for this example to live, thoughts?

**Comment by zachgk** [comment]
@tomoncle @git-ness https://github.com/apache/incubator-mxnet/blob/master/example/README.md seems like a good place to put it. I think it fits under "Web Predictive Services". Would you like to raise a PR to add it?
